android/java/org/chromium/chrome/browser/vpn/split_tunnel/SplitTunnelActivity.java

```diff
@@ -0,0 +1,184 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.vpn.split_tunnel;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.widget.NestedScrollView;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.firstrun.BraveFirstRunFlowSequencer;
+import org.chromium.chrome.browser.init.AsyncInitializationActivity;
+import org.chromium.chrome.browser.vpn.BraveVpnNativeWorker;
+import org.chromium.chrome.browser.vpn.utils.BraveVpnPrefUtils;
+import org.chromium.chrome.browser.vpn.utils.BraveVpnUtils;
+
+public class SplitTunnelActivity extends AsyncInitializationActivity
+        implements LifecycleOwner, ApplicationListAdapter.OnApplicationClickListener {
+    private BraveFirstRunFlowSequencer mFirstRunFlowSequencer;
+    private RecyclerView mRecyclerViewSystemApps;
+    private ApplicationListAdapter mRecyclerViewAdapterExcludedApps;
+    private ApplicationListAdapter mRecyclerViewAdapterApps;
+    private ApplicationListAdapter mRecyclerViewAdapterSystemApps;
+
+    private void initializeViews() {
+        setContentView(R.layout.activity_split_tunnel);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        assert actionBar != null;
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setTitle(getResources().getString(R.string.split_tunneling));
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
+
+        DividerItemDecoration dividerItemDecoration =
+                new DividerItemDecoration(SplitTunnelActivity.this, LinearLayoutManager.VERTICAL);
+
+        RecyclerView recyclerViewExcludedApps = findViewById(R.id.rv_excluded_apps);
+        recyclerViewExcludedApps.setLayoutManager(linearLayoutManager);
+        recyclerViewExcludedApps.addItemDecoration(dividerItemDecoration);
+        mRecyclerViewAdapterExcludedApps = new ApplicationListAdapter(this, true);
+        recyclerViewExcludedApps.setAdapter(mRecyclerViewAdapterExcludedApps);
+
+        LinearLayoutManager linearLayoutManager2 = new LinearLayoutManager(this);
+        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
```

Comment:
shouldn't we make a util method create layoutmanger and assign it to RV to avoid repetition in code?

Reply:
i have updated .xml with app:layoutManager="LinearLayoutManager"]